File: Program/php/MotorSport.php
<?php
require_once "Motor.php";

// Motor (Multilevel Inheritance)
class MotorSport extends Motor {
    private $kecepatanMaksimal;
    private $harga;
    private $sistemPengereman;

    public function __construct($id = 0, $merk = "", $tahun = 0, $foto = "", $kapasitasMesin = 0, $jenisTransmisi = "", $kapasitasTangki = 0.0, $kecepatanMaksimal = 0, $harga = 0.0, $sistemPengereman = "") {
        parent::__construct($id, $merk, $tahun, $foto, $kapasitasMesin, $jenisTransmisi, $kapasitasTangki);
        $this->kecepatanMaksimal = $kecepatanMaksimal;
        $this->harga = $harga;
        $this->sistemPengereman = $sistemPengereman;
    }

    public function getKecepatanMaksimal() { return $this->kecepatanMaksimal; }
    public function setKecepatanMaksimal($kecepatanMaksimal) { $this->kecepatanMaksimal = $kecepatanMaksimal; }

    public function getHarga() { return $this->harga; }
    public function setHarga($harga) { $this->harga = $harga; }

    public function getSistemPengereman() { return $this->sistemPengereman; }
    public function setSistemPengereman($sistemPengereman) { $this->sistemPengereman = $sistemPengereman; }
}
?>

File: Program/c++/MotorSport.cpp
#include <iostream>
#include "Motor.cpp"

// Subclass dari Motor / multilevel inheritance
class MotorSport : public Motor {
    private:
        int kecepatanMaksimal; // km/h
        double harga;           // Rp
        string sistemPengereman;

    public:
        // Default Constructor
        MotorSport() : Motor() {
            this->kecepatanMaksimal = 0;
            this->harga = 0.0;
            this->sistemPengereman = "";
        }

        // Constructor dengan Parameter
        MotorSport(int id, string merk, int tahun, int kapasitasMesin, string jenisTransmisi, double kapasitasTangki, int kecepatanMaksimal, double harga, string sistemPengereman) : Motor(id, merk, tahun, kapasitasMesin, jenisTransmisi, kapasitasTangki) {
            this->kecepatanMaksimal = kecepatanMaksimal;
            this->harga = harga;
            this->sistemPengereman = sistemPengereman;
        }

        // Getter dan Setter Kecepatan Maksimal
        void setKecepatanMaksimal(int kecepatanMaksimal) { this->kecepatanMaksimal = kecepatanMaksimal; }
        int getKecepatanMaksimal() { return this->kecepatanMaksimal; }

        // Getter dan Setter Harga
        void setHarga(double harga) { this->harga = harga; }
        double getHarga() { return this->harga; }

        // Getter dan Setter Sistem Pengereman
        void setSistemPengereman(string sistemPengereman) { this->sistemPengereman = sistemPengereman; }
        string getSistemPengereman() { return this->sistemPengereman; }

        // Destructor
        ~MotorSport() {}
};